src/test/java/de/tum/in/www1/artemis/util/CourseTestService.java

```diff
@@ -1552,4 +1552,147 @@ public void testGetExerciseStatsForCourseOverviewWithPastExercises() throws Exce
         var statisticsOptional = exerciseDTOS.stream().filter(exercise -> exercise.getExerciseId().equals(exerciseId)).findFirst();
         assertThat(statisticsOptional.isEmpty()).isTrue();
     }
+
+    // Test
+    public void testGetCourseManagementDetailData() throws Exception {
+        ZonedDateTime now = ZonedDateTime.now();
+        // add courses with exercises
+
+        var course = database.createCoursesWithExercisesAndLectures(true).get(0);
+
+        var student1 = ModelFactory.generateActivatedUser("user1");
+        var student2 = ModelFactory.generateActivatedUser("user2");
+        // Fetch and update an instructor
+        var instructor = database.getUserByLogin("instructor1");
+        var groups = new HashSet<String>();
+        groups.add("instructor");
+        instructor.setGroups(groups);
+
+        userRepo.save(instructor);
+        userRepo.save(student1);
+        userRepo.save(student2);
+
+        var releaseDate = now.minusDays(7);
+        var dueDate = now.minusDays(2);
+        var assessmentDueDate = now.minusDays(1);
+        var exercise = ModelFactory.generateTextExercise(releaseDate, dueDate, assessmentDueDate, course);
+        exercise.setMaxPoints(5.0);
+        exercise = exerciseRepo.save(exercise);
+
+        // Add a single participation to that exercise
+        final var exerciseId = exercise.getId();
+
+        var result1 = database.createParticipationSubmissionAndResult(exerciseId, student1, 5.0, 0.0, 60, true);
+        var result2 = database.createParticipationSubmissionAndResult(exerciseId, student2, 5.0, 0.0, 40, true);
```

Comment:
More vars that at least I don't really like...

Reply:
In this case i would say that the variable names describe the variable well enough. `Result result1` somehow sounds duplicated, doesn't it?

Reply:
Can be discussed in the next meeting